external/fv3fit/fv3fit/reservoir/reservoir.py

```diff
@@ -0,0 +1,120 @@
+import dataclasses
+import logging
+import numpy as np
+import scipy
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class ReservoirHyperparameters:
+    """Hyperparameters for reservoir
+
+    input_dim: Size of input vector
```

Comment:
nit: we usually use `dim` to mean the name of a dimension. E.g. see a lot of our VCM code or xarray convention. So I'd suggest renaming to `input_size` and same for `reservoir_state_dim` and `output_dim`.